jetbrains-core/src/software/aws/toolkits/jetbrains/services/codewhisperer/toolwindow/CodeWhispererCodeReferenceManager.kt

```diff
@@ -70,8 +70,10 @@ class CodeWhispererCodeReferenceManager(private val project: Project) {
         val startOffset = caretPosition.offset
         val relativePath = getRelativePathToContentRoot(editor)
         reformattedDetail.references().forEachIndexed { i, reference ->
+            val recommendationEnd = startOffset + reformattedDetail.content().length
             val start = startOffset + reference.recommendationContentSpan().start()
-            val end = startOffset + reference.recommendationContentSpan().end()
+            val tempEnd = startOffset + reference.recommendationContentSpan().end()
+            val end = if (tempEnd <= recommendationEnd) tempEnd else recommendationEnd
```

Comment:
This is not needed if the reference truncation logic is handled before correctly. The reference.recommendationContentSpan().start() is guaranteed to be smaller than the recommendation length.